src/mmrelay/matrix_utils.py

```
@@ -1546,137 +1698,255 @@ async def connect_matrix(
+                        await asyncio.to_thread(
+                            save_credentials,
+                            auth_info.credentials,
+                            credentials_path=auth_info.credentials_path,
+                        )
+                        logger.info(
+                            "Updated credentials.json with discovered information"
+                        )
                     except OSError as e:
-                        logger.warning(f"Failed to persist discovered device_id: {e}")
+                        logger.exception(
+                            f"Failed to persist updated session information: {e}"
```

Ruff (0.14.14)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)